types/interface/link/link.pp

```diff
@@ -0,0 +1,84 @@
+# interface definition
+type Systemd::Interface::Link::Link = Struct[{
+    'Description'                         => Optional[String[1]],
```

Comment:
```suggestion
    Optional['Description']                         => String[1],
```

Does this stop a massive hash of unset values being added to the catalogue.

Reply:
Oh interesting did not realize this difference yet. Do you have more information about the difference ?

Reply:
I don't have a reference.

One says the key does not have to be present in the hash at all.

The other adds the key with a default value or undef. 
Would have to check the catalogue to see if there is a difference.

Reply:
Theres no undef in the catalog either way.
Ading the following type:
```
type Test::Test = Struct[{
    'always'        => String[1],
    'one'           => Optional[String[1]],
    Optional['two'] => String[1],
}]
```
Using it in a class:
`Test::Test $te   = { 'always' => '2' },`

Renders in the catalog to:

```
- type: Class
  ....
  parameters:
    te:
      always: '2'
```
Neither `one` nor `two` key of the hash is in the catalog.
So I assume there is no difference in the result beside the writing in the code.